sympy/matrices/sparse.py

```diff
@@ -403,6 +408,10 @@ def _eval_transpose(self):
     def _eval_values(self):
         return [v for k,v in self._smat.items() if not v.is_zero]
 
+    @classmethod
+    def _eval_zeros(cls, rows, cols):
```

Comment:
`zeros` is defined for all matrices and does error checking and has a consistent doc string.  `_eval_zeros()` does the real work and is mean to be overridden in subclasses.